suite-native/module-device-settings/src/hooks/useDeviceConnectionGuard.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectIsDeviceConnected } from '@suite-common/wallet-core';
+import {
+    AuthorizeDeviceStackRoutes,
+    DeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    SettingsStackParamList,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SettingsStackParamList,
+    DeviceStackRoutes.DeviceSettings,
+    RootStackParamList
+>;
+
+export const useDeviceConnectionGuard = () => {
+    const navigation = useNavigation<NavigationProps>();
+
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+
+    const redirectToConnectAndUnlockScreen = useCallback(() => {
+        navigation.navigate(RootStackRoutes.AuthorizeDeviceStack, {
+            screen: AuthorizeDeviceStackRoutes.ConnectAndUnlockDevice,
+            params: {
+                onCancelNavigationTarget: {
+                    name: RootStackRoutes.DeviceSettingsStack,
+                    params: { screen: DeviceStackRoutes.DeviceSettings },
+                },
+            },
+        });
+    }, [navigation]);
+
+    useEffect(() => {
+        if (!isDeviceConnected) return redirectToConnectAndUnlockScreen();
```

Comment:
Device is disconnected during FW installation flow because it has to be restarted to bootloader mode and this effect is active there. 
![Uploading image.png…]()

Reply:
https://github.com/user-attachments/assets/6170ab53-5146-4a89-a317-92567a6ed1b2

Reply:
ok that's new 😅 on it!

Reply:
https://github.com/trezor/trezor-suite/pull/17545/commits/1e4fe01caf37d78ed486a1b87d5ee1495204d282 

https://github.com/user-attachments/assets/1227ce37-3119-4790-94ec-ddb20711f829